### DISCUSSION TITLE
Clicking button in iframe does not add additional options to another dropdown in the iframe

### DISCUSSION THREAD

**Discussion by tausif29** [cypress-io/cypress#25697] [Questions and Help]
![image](https://user-images.githubusercontent.com/39052801/216571370-4e01746c-448d-4a71-a573-fcbab3323c0e.png)

As per the screenshot, We have a toolbar in an iframe,  We have one + (Add) button, which adds additional options to the highlighted dropdown, It works fine manually, but does not work with cypress test, I am able to click on the + button inside the iframe, But its adding require options to the dropdown.

**Comment by portyu9** [comment]
Hi @tausif29 

The iframe access is working—the screenshot shows Cypress finds and clicks the correct `createNewMarkup` button.

The issue is likely that this viewer requires a **native/trusted mouse event**. Cypress `.click()` generates simulated events (`event.isTrusted === false`), while a manual click generates a trusted browser event.

Try a native click with `cypress-real-events` instead of `.trigger('mouseover').click({ force: true })`:

```js
import 'cypress-real-events'

cy.get('iframe')
  .its('0.contentDocument.body')
  .should('not.be.empty')
  .then(cy.wrap)
  .find('button[data-element="createNewMarkup"]')
  .should('be.visible')
  .realClick()
```

Also remove `{ force: true }` unless it is genuinely necessary.

Once the real click puts the viewer into markup mode, the `Annotate`, `Shapes`, and `Insert` options should be added to the dropdown.

**Reply by portyu9** [reply]
@tausif29 Please mark this answer as accepted if it helped, thank you.
